deepchecks/tabular/checks/data_integrity/mixed_nulls.py

```diff
@@ -84,13 +86,26 @@ def run_logic(self, context: Context, dataset_kind) -> CheckResult:
         result_dict = {}
 
         for column_name in list(df.columns):
-            column_data = df[column_name]
-
-            string_null_counts = {value: count for value, count in column_data.value_counts(dropna=True).iteritems()
-                                  if string_baseform(value) in null_string_list}
+            column_data = (
+                # NOTE:
+                # `pandas.Series.value_counts` and 'pandas.Series.apply'
+                # work in an unusual way with categorical data types
+                # - `value_counts` returns all categorical values even if they are not in series
+                # - `apply` applies function to each category, not to values
+                # therefore we need cast it
+                # NOTE:
+                # `Series.astype('object')` will transform `numpy.nan` into `math.nan`
```

Comment:
I'm not sure we are okay with this, we want the output to represent the real nans of the user. As an alternative we just use `value_counts()` and filter all the values with 0

Reply:
agree (I changed implementation)
but `value_counts` also might transform `np.nan` into `math.nan`

![image](https://user-images.githubusercontent.com/71635444/175335943-4f719538-38be-49eb-a7bc-926346a84823.png)

Reply:
note: this does not happen with `name` column, at the end, it is `np.nan`

Reply:
whenever you try to iterate over the series, pandas transforms `np,nan` into `math.nan`

![image](https://user-images.githubusercontent.com/71635444/175339907-e76ced33-d01a-46da-9fdf-2ef3a368d12d.png)